pastas/model.py

```diff
@@ -1,25 +1,36 @@
-"""This module contains the Model class in Pastas."""
+#  This module contains the Model class in Pastas.
 
+# Python Dependencies
 from collections import OrderedDict
 from itertools import combinations
 from logging import getLogger
 from os import getlogin
 
+# External Dependencies
 import numpy as np
 from pandas import (DataFrame, Series, Timedelta, Timestamp,
-                    date_range, concat, to_timedelta)
-
-from .decorators import get_stressmodel
-from .io.base import _load_model, dump
-from .modelstats import Statistics
-from .noisemodels import NoiseModel
-from .modelplots import Plotting
-from .solver import LeastSquares
-from .stressmodels import Constant
-from .timeseries import TimeSeries
-from .utils import (_get_dt, _get_time_offset, frequency_is_supported,
-                    get_sample, validate_name)
-from .version import __version__
+                    date_range, concat, to_timedelta, DatetimeIndex)
+
+# Internal Pastas
+from pastas.decorators import get_stressmodel
+from pastas.io.base import _load_model, dump
+from pastas.modelstats import Statistics
+from pastas.noisemodels import NoiseModel
+from pastas.modelplots import Plotting
+from pastas.solver import LeastSquares
+from pastas.stressmodels import Constant
+from pastas.timeseries import TimeSeries
+from pastas.transform import ThresholdTransform
+from pastas.utils import (_get_dt, _get_time_offset, frequency_is_supported,
+                          get_sample, validate_name)
+from pastas.version import __version__
+
+# Type Hinting
+from typing import Union, Optional, Tuple, List
+from pastas.typing import StressModel, Solver, Tminmax, Array_Like
```

Comment:
Should we just label Tmintmax as a Timestamp?

Reply:
See comment below on types.py